library/alloc/src/collections/btree/node.rs

```diff
@@ -600,9 +600,6 @@ impl<K, V> NodeRef<marker::Owned, K, V, marker::LeafOrInternal> {
     /// no cleanup is done on any of the keys, values and other children.
     /// This decreases the height by 1 and is the opposite of `push_internal_level`.
     ///
-    /// Requires exclusive access to the `NodeRef` object but not to the root node;
-    /// it will not invalidate other handles or references to the root node.
```

Comment:
This is definitely confusing, but I think this must be referring to the *new* root node, i.e. that first (and only) child that gets assigned to `self.node`. That would be relevant to the `ExtractIfInner` and `CursorMutKey` callers, which have a `NodeRef` separate from the one that is getting popped. (So that separate ref must be the child that will be the new root.)

If you agree, can you try to clarify this comment instead of removing it?

Reply:
Good point! I made an attempt to do so.